Tribler/community/doubleentry/community.py

```diff
@@ -0,0 +1,325 @@
+        # TODO
+        up = 1
+        down = 2
+        total_up = 3
+        total_down = 4
+        # Instantiate the personal information
+        sequence_number_requester = self.persistence.get_latest_sequence_number(self._public_key)
+        previous_hash_requester = self.persistence.get_previous_id()
+        public_key_requester = self._public_key
+
+        data_without_signature = (up, down, total_up, total_down,
+                                  sequence_number_requester, previous_hash_requester, public_key_requester)
+        # Create the signature
+        signature = ECCrypto().create_signature(self._ec, encode_signing_format(data_without_signature))
+
+        message = meta.impl(authentication=(self.my_member,),
+                            distribution=(self.claim_global_time(),),
+                            destination=(candidate,),
+                            # Append the signature to the data.
+                            payload=data_without_signature + (signature,))
+        return message
+
+    def validate_signature_request(self, signature_request):
+        """
+        Validates if the signature_request message contains correct signatures and public keys.
+        :param signature_request: The message that needs to be checked.
+        :return: Boolean containing the truth value of the validity of the message.
+        """
+        payload = signature_request.payload
+        # Check if the request is not already processed.
+        return (not self.persistence.contains_signature(payload.signature_requester, payload.public_key_requester) and
+                # Check if the request is valid
+                self.validate_signature(payload.public_key_requester, payload.signature_data_requester(),
+                                        payload.signature_requester))
+
+    def _check_signature_request(self, messages):
+        """
+        Generator that generates a list of correct signature_request messages from a list of messages.
+        :param messages: Potential valid signature_request messages.
+        :return: Series of correct signature_request messages
+        """
+        self._logger.info("Received " + str(len(messages)) + " signature requests.")
+        for message in messages:
+            if self.validate_signature_request(message):
+                self._logger.info("Received valid request.")
+                yield message
+            else:
+                DropMessage(message, "Invalid signature request message")
+
+    def _on_signature_request(self, messages):
+        """
+        Handles behaviour of the community when it receives a signature_request message.
+        It will send out a signature response
+        :param messages: Signature_request messages that needs to be handled.
+        """
+        for message in messages:
+            # Check if the messages are not from ourselves.
+            if message.payload.public_key_requester != self._public_key:
```

Comment:
This check needs to be performed in the check, and if it fails yield a DropMessage